bw2data/backends/iotable/__init__.py

```diff
@@ -48,19 +50,48 @@ def write_exchanges(self, technosphere, biosphere, dependents):
             name=clean_datapackage_name(self.name + " inventory geomapping matrix"),
             nrows=len(self),
         )
+
+
         print("Adding technosphere matrix")
-        dp.add_persistent_vector_from_iterator(
-            matrix="technosphere_matrix",
-            name=clean_datapackage_name(self.name + " technosphere matrix"),
-            dict_iterator=technosphere,
-        )
+        # if technosphere is a dictionary pass it's keys & values
+        if isinstance(technosphere, dict):
+            dp.add_persistent_vector(
+                matrix="technosphere_matrix",
+                name=clean_datapackage_name(self.name + " technosphere matrix"),
+                **technosphere,
+            )
+        # if it is an iterable, convert to right format
+        elif hasattr(technosphere, '__iter__'):
+            dp.add_persistent_vector_from_iterator(
+                matrix="technosphere_matrix",
+                name=clean_datapackage_name(self.name + " technosphere matrix"),
+                dict_iterator=technosphere,
+            )
+        else:
+            raise Exception(f"Error: Unsupported technosphere type: {type(technosphere)}")
+
 
         print("Adding biosphere matrix")
-        dp.add_persistent_vector_from_iterator(
-            matrix="biosphere_matrix",
-            name=clean_datapackage_name(self.name + " biosphere matrix"),
-            dict_iterator=biosphere,
-        )
+        # if biosphere is a dictionary pass it's keys & values
+        if isinstance(biosphere, dict):
+            dp.add_persistent_vector(
+                matrix="biosphere_matrix",
+                name=clean_datapackage_name(self.name + " biosphere matrix"),
+                **biosphere,
+            )
+        # if it is an iterable, convert to right format
+        elif hasattr(biosphere, '__iter__'):
+            dp.add_persistent_vector_from_iterator(
+                matrix="biosphere_matrix",
+                name=clean_datapackage_name(self.name + " biosphere matrix"),
+                dict_iterator=biosphere,
+            )
+        else:
+            raise Exception(f"Error: Unsupported biosphere type: {type(technosphere)}")
+
+
+        # finalize
+        print("Finalizing serialization")
```

Comment:
@BenPortner , it would be better if you used a logger instead of direct "print" statements.
```python
import logging
logger = logging.getLogger(__name__)
...
logger.info("Finalizing serialization")
```
I think loggers are not used enough in general, so we better start using them on the codebase.

Reply:
@tngTUDOR Respectfully disagree. Logging could be in addition to the print statement, but this is useful feedback for less experienced people wondering if the command they just ran is actually doing something.

Reply:
I understand, having feedback is absolutely necessary. That is the mission of `logging.info` (just like `logging.error` has as mission presenting information with a different format to attire the attention of the user). 
loggers can be equivalent to print, when configured to do so (no module name in the printed string, no date time).